src/Explorer/Controls/NotebookGallery/GalleryViewerComponent.tsx

```diff
@@ -134,9 +139,17 @@ export class GalleryViewerComponent extends React.Component<GalleryViewerCompone
     const tabs: GalleryTabInfo[] = [this.createTab(GalleryTab.OfficialSamples, this.state.sampleNotebooks)];
 
     if (this.props.container?.isGalleryPublishEnabled()) {
-      tabs.push(this.createTab(GalleryTab.PublicGallery, this.state.publicNotebooks));
+      tabs.push(
+        this.createPublicGalleryTab(
+          GalleryTab.PublicGallery,
+          this.state.publicNotebooks,
+          this.state.acceptedCodeOfConduct
+        )
+      );
       tabs.push(this.createTab(GalleryTab.Favorites, this.state.favoriteNotebooks));
-      tabs.push(this.createTab(GalleryTab.Published, this.state.publishedNotebooks));
+      if (this.state.acceptedCodeOfConduct !== false) {
```

Comment:
Remove the `!== false` part (it's a boolean, right?)

Reply:
So the reason I check for false is, if its is undefined to start with when galleryViewerComponent is loaded and i check for (!this.state.acceptedCodeOfConduct), then the codeOfConduct component is always shown first and then removed to make way for the gallery (in case the juno call to read codeOfConduct flag returns true). So for a few milliseconds, the code of conduct component will pop up even if it has been formerly accepted. The experience looked bad, so I resorted to this type of check.

Reply:
I see. Then I would opt for something more explicit:
`this.state.acceptedCodeOfConduct !== undefined && !this.state.acceptedCodeOfConduct`
Or add a comment about why you're using this construct.

Reply:
will add a comment